boards/airbourne/airbourne_board.cpp

```diff
@@ -73,32 +75,49 @@ void AirbourneBoard::clock_delay(uint32_t milliseconds)
 }
 
 // serial
-void AirbourneBoard::serial_init(uint32_t baud_rate)
+void AirbourneBoard::serial_init(uint32_t baud_rate, uint32_t dev)
 {
-  (void)baud_rate;
   vcp_.init();
+  if (dev == 3)
+  {
+    uart3_.init(&uart_config[UART3], baud_rate);
+    current_serial_ = &uart3_;
+    secondary_serial_device_ = dev;
+  }
+  else
+  {
+    secondary_serial_device_ = dev;
+    current_serial_ = &vcp_;
+  }
 }
 
 void AirbourneBoard::serial_write(const uint8_t *src, size_t len)
 {
-  vcp_.write(src, len);
+  current_serial_->write(src, len);
 }
 
 uint16_t AirbourneBoard::serial_bytes_available()
 {
-  return vcp_.rx_bytes_waiting();
+  if (vcp_.connected())
```

Comment:
The logic seems to be slightly obfuscated in this implementation. If I'm correct, the big-picture logic is:

  * Use VCP if connected
  * If VCP is not connected _and_ another serial device has been set as the desired device, then use that device

I think we could streamline the implementation a bit to make this more clear?

Reply:
Yes, that is correct.  That would be better